From dcc68687d0809b452a1532e351348e2a5456d61b Mon Sep 17 00:00:00 2001
From: Carlos Munoz <cmunoz@caviumnetworks.com>
Date: Mon, 7 Jul 2014 18:05:50 -0700
Subject: [PATCH 924/974] netdev: octeon3-ethernet: Only swap pki buffer links
 on cn78xx pass 1.

Signed-off-by: Carlos Munoz <cmunoz@caviumnetworks.com>
[Original patch taken from Cavium SDK 3.1.2-568]
Signed-off-by: Quanyang Wang <quanyang.wang@windriver.com>
---
 drivers/net/ethernet/octeon/octeon3-ethernet.c | 12 ++++++++----
 1 file changed, 8 insertions(+), 4 deletions(-)

diff --git a/drivers/net/ethernet/octeon/octeon3-ethernet.c b/drivers/net/ethernet/octeon/octeon3-ethernet.c
index ebb1c2a..c3eabe1 100644
--- a/drivers/net/ethernet/octeon/octeon3-ethernet.c
+++ b/drivers/net/ethernet/octeon/octeon3-ethernet.c
@@ -1133,8 +1133,10 @@ static int octeon3_eth_rx_one(struct octeon3_rx *rx, bool is_async,
 				break;
 			packet_ptr.u64 = *(u64 *)(data - 8);
 #ifndef __LITTLE_ENDIAN
-			/* PKI_BUFLINK_S's are endian-swapped */
-			packet_ptr.u64 = swab64(packet_ptr.u64);
+			if (OCTEON_IS_MODEL(OCTEON_CN78XX_PASS1_X)) {
+				/* PKI_BUFLINK_S's are endian-swapped */
+				packet_ptr.u64 = swab64(packet_ptr.u64);
+			}
 #endif
 			data = phys_to_virt(packet_ptr.addr);
 			skb = octeon3_eth_work_to_skb((void *)round_down((unsigned long)data, 128ul));
@@ -1172,8 +1174,10 @@ static int octeon3_eth_rx_one(struct octeon3_rx *rx, bool is_async,
 				break;
 			packet_ptr.u64 = *(u64 *)(data - 8);
 #ifndef __LITTLE_ENDIAN
-			/* PKI_BUFLINK_S's are endian-swapped */
-			packet_ptr.u64 = swab64(packet_ptr.u64);
+			if (OCTEON_IS_MODEL(OCTEON_CN78XX_PASS1_X)) {
+				/* PKI_BUFLINK_S's are endian-swapped */
+				packet_ptr.u64 = swab64(packet_ptr.u64);
+			}
 #endif
 			data = phys_to_virt(packet_ptr.addr);
 			next_skb = octeon3_eth_work_to_skb((void *)round_down((unsigned long)data, 128ul));
-- 
2.6.2

